### DISCUSSION TITLE
Can the manager version be forced?

### DISCUSSION THREAD

**Discussion by deronnax** [renovatebot/renovate#25281] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

36.109.4

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Hello. 
Pipenv regularly break features and it's not uncommon new releases don't work with Pipfile the previous releases used to. Renovate seem to always use the last version of Pipenv. Is it possible to force renovate to use a given version of a manager?
Cheers.

### What you tried so far.

nothing

**Comment by rarkins** [comment, marked as answer]
Yes, using `constraints` if you need to manually specify pipenv version to use. Otherwise Renovate tries to detect the version to use here: https://github.com/renovatebot/renovate/blob/a1e5f548d148f96904ca505c08a4c11825f98446/lib/modules/manager/pipenv/artifacts.ts#L51-L79

**Reply by deronnax** [reply]
han, I saw it in the code but I wasn't sure. The documentation could be improved : the python page mention constraints but it seems to apply only to python itself: https://docs.renovatebot.com/modules/manager/pip-compile/#configuration-of-python-version
Thank you very much for your fast answer.

**Reply by rarkins** [reply]
1. That's not "the python page", it's the pip-compile manager page and not related to the poetry manager you are using
2. The `constraints` feature is documented separately and lists `poetry` as supported: https://docs.renovatebot.com/configuration-options/#constraints
3. You are still in violation of https://github.com/renovatebot/renovate/blob/main/docs/development/help-us-help-you.md#avoid-complaining-about-docs-after-getting-help

Because you are so insistent on still retaining a complaint about the documentation after your edit, I will be insistent that you now contribute a PR improving it. If you think it would be best to mention it in the poetry documentation, the place to make the edit/addition would be here: https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/poetry/readme.md
